internal/schema/0.14/terraform.go

```diff
@@ -26,20 +26,10 @@ func terraformBlockSchema(v *version.Version) *schema.BlockSchema {
 						"with this configuration, e.g. `~> 0.12`"),
 				},
 				"experiments": {
-					Constraint: schema.Set{
-						Elem: schema.OneOf{
-							schema.Keyword{
-								Keyword: "module_variable_optional_attrs",
-								Name:    "feature",
-							},
-							schema.Keyword{
-								Keyword: "provider_sensitive_attrs",
-								Name:    "feature",
-							},
-						},
-					},
-					IsOptional:  true,
-					Description: lang.Markdown("A set of experimental language features to enable"),
+					Constraint: schema.LiteralType{Type: cty.String},
```

Comment:
I'm not sure what the best constraint here is, since we don't have "any keyword" constraint but I'd worry that this will cause some noise during validation because keywords are not quoted strings.

Reply:
Good point. Should we remove the constraint entirely? Or is there something else you're hinting at?

Reply:
In the lack of `AnyKeyword` I don't know honestly. Perhaps implementing such a constraint is the best course of action here. I expect it would be basically a copy of [`schema.Keyword`](https://pkg.go.dev/github.com/hashicorp/hcl-lang@v0.0.0-20240326153306-49d737897778/schema#Keyword) and then you would use it like this:

```
Constraint: schema.Set{
	Elem: schema.AnyKeyword{Description: "See active experiments at https://..."},
},
```

In practice I imagine you'd be looking at https://github.com/hashicorp/hcl-lang/blob/49d737897778/schema/constraint_keyword.go#L14 and https://github.com/hashicorp/hcl-lang/blob/main/decoder/expr_keyword.go#L11 and any occurrences of that throughout the codebase. Maybe more work than you were hoping for, but I cannot think of a quick and easy solution here. That doesn't mean there isn't one - I just cannot think of it.

--- 

Relatedly - whatever constraint you choose here I think it should remain wrapped in `schema.Set{}` as that represents the `[]`, i.e. that there can be more than one experiment active at any time.

Reply:
@dbanck I'm unsure who that question is aimed at but that seems reasonable to me! 😄

Reply:
Oh, sorry. That was directed at @jpogran as a suggestion to get this PR merged before release.

Reply:
added a commit to just remove `module_variable_optional_attrs` starting in Terraform 1.8 – as Daniel said, we can revisit this later and figure out how to define something when there are no active experiments 👍